src/parser/FStar.Parser.Dep.fs

```diff
@@ -502,9 +510,6 @@ let collect_one
         match resolve_module_name original_or_working_map key with
         | Some module_name ->
           add_dep deps (dep_edge module_name);
-          if has_interface original_or_working_map module_name
```

Comment:
We don't need to add to `mo_roots` here, since when an `.fsti` is processed, its corresponding `.fst` is added there already.